### PR TITLE
Feature: convert to Preact, add api, and allow non-Odyssey videos

### PR DESCRIPTION
**Pull request by AshKyd**

### CODE REVIEW

**Review by phocks — APPROVED** [review]
Looks good!

File: src/components/App/index.tsx [hunk 1]
```diff
@@ -215,22 +154,29 @@ const App = () => {
...
       </div>
 
-      <div className={`${styles.text} ${isMuted && styles.hidden}`}>
+      <div className={`${styles.text} ${styles.textContinueReading} ${isMuted && styles.hidden}`}>
         Keep scrolling to read the story
       </div>
-      <div className={`${styles.text} ${!isMuted && styles.hidden}`}>
-        This story is best experienced with sound on
+      <div className={`${styles.text} ${styles.textSoundOn} ${!isMuted && styles.hidden}`}>
+        This story is best<br />experienced with sound on!!!!!
```
Comment (drzax) [inline src/components/App/index.tsx]: Do we need this many exclamation points?

File: src/components/App/utils.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
+// How many seconds before we unload videos
+const SECONDS_BEFORE_UNLOAD = 30;
+
+function playVideo(video) {
+  if (video.api) {
+    return video.api.play();
```
Comment (drzax) [inline src/components/App/utils.js:6]: I think this is fine as is, but I wonder if the conditionals are necessary, could we just ignore the API prop or do we need to interact with Odyssey videos that way to keep the API in sync with reality?

**Review by drzax — APPROVED** [review]
LGTM

File: src/components/App/index.tsx (hunk 1 above)
Comment (AshKyd) [reply src/components/App/index.tsx]: hahahaha no, that was just so I could test it. Good catch 😬

File: src/components/App/utils.js (hunk 2 above)
Comment (phocks) [reply src/components/App/utils.js:6]: From memory we do want to interact with the Odyssey videos and keep everything in sync globally if they are Odyssey videos.. but I can't remember why. If it still works directly with video.play then maybe it's not needed?

Comment (drzax) [reply src/components/App/utils.js:6]: I'd suggest we leave it as is for now, but a possible candidate for future simplification.

**Pull request merged**